routes/v1.js

```diff
@@ -456,39 +465,49 @@ api.declare({
     return;
   }
 
+  // Parse timestamps
+  let created = new Date(taskDef.created);
+  let deadline = new Date(taskDef.deadline);
+  let expires = new Date(taskDef.expires);
+
   // Insert entry in deadline queue
-  var deadline = new Date(taskDef.deadline);
   await this.queueService.putDeadlineMessage(taskId, deadline);
 
   // Try to create Task entity
   try {
+    let runs = [];
+    // Add run if there is no dependencies
+    if (taskDef.dependencies.length === 0) {
+      runs.push({
+        state:            'pending',
+        reasonCreated:    'scheduled',
+        scheduled:        new Date().toJSON(),
+      });
+    }
     var task = await this.Task.create({
-      taskId:           taskId,
-      provisionerId:    taskDef.provisionerId,
-      workerType:       taskDef.workerType,
-      schedulerId:      taskDef.schedulerId,
-      taskGroupId:      taskDef.taskGroupId,
-      routes:           taskDef.routes,
-      priority:         taskDef.priority,
-      retries:          taskDef.retries,
-      retriesLeft:      taskDef.retries,
-      created:          new Date(taskDef.created),
-      deadline:         deadline,
-      expires:          new Date(taskDef.expires),
-      scopes:           taskDef.scopes,
-      payload:          taskDef.payload,
-      metadata:         taskDef.metadata,
-      tags:             taskDef.tags,
-      extra:            taskDef.extra,
-      runs:             [{
-        state:          'pending',
-        reasonCreated:  'scheduled',
-        scheduled:      new Date().toJSON()
-      }],
-      takenUntil:       new Date(0)
+      taskId:             taskId,
+      provisionerId:      taskDef.provisionerId,
+      workerType:         taskDef.workerType,
+      schedulerId:        taskDef.schedulerId,
+      taskGroupId:        taskDef.taskGroupId,
+      dependencies:       taskDef.dependencies,
+      dependencyRelation: taskDef.dependencyRelation,
+      routes:             taskDef.routes,
+      priority:           taskDef.priority,
+      retries:            taskDef.retries,
+      retriesLeft:        taskDef.retries,
+      created:            created,
+      deadline:           deadline,
+      expires:            expires,
+      scopes:             taskDef.scopes,
+      payload:            taskDef.payload,
+      metadata:           taskDef.metadata,
+      tags:               taskDef.tags,
+      extra:              taskDef.extra,
+      runs:               runs,
+      takenUntil:         new Date(0),
     });
-  }
-  catch (err) {
+  } catch (err) {
     // We can handle cases where entity already exists, not that, we re-throw
     if (!err || err.code !== 'EntityAlreadyExists') {
```

Comment:
it's just an old leftover in case we somewhere did `throw null`... it shouldn't happen, but with promises sometime a promise can be rejected without an error...

I don't think we have to do this everywhere.. I think I was just being very defensive when I wrote this :)